Common/Common.DataBaseRepositories/Repository.cs

```diff
@@ -0,0 +1,281 @@
+﻿// // <copyright company="Tarcha & Ivchenko Company">
+// //      Copyright (c) 2015, All Right Reserved
+// // </copyright>
+// // <author>Ivan Ivchenko</author>
+// // <author>Myroslava Tarcha</author>
+
+using System;
+using System.Collections.Generic;
+using System.Data.Entity;
+using System.Data.Entity.Core;
+using System.Data.Entity.Infrastructure;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Linq.Expressions;
+using NLog;
+
+namespace DataBaseRepositories
+{
+    /// <summary>
+    /// Provides base methods for working with entities.
+    /// </summary>
+    /// <typeparam name="TEntity">The mapped type.</typeparam>
+    /// <typeparam name="TContext">Type of the used DbContext.</typeparam>
+    public class Repository<TEntity, TContext> : IRepository<TEntity>, IDisposable
```

Comment:
- [ ] Lets mark the class as sealed, delete Logger property and DataContext property. Lets logger name will be repository+entity